src/engine/openspaceengine.cpp

```diff
@@ -497,22 +499,124 @@ void OpenSpaceEngine::initialize() {
     // Load a light and a monospaced font
     loadFonts();
 
-    // Initialize the Scene
-    // @CLEANUP:  This should become a unique_ptr that is either created inside the
-    // renderengine or moved into it ---abock
-    Scene* sceneGraph = new Scene;
-    sceneGraph->initialize();
-    
     std::string scenePath = "";
     configurationManager().getValue(ConfigurationManager::KeyConfigScene, scenePath);
-    sceneGraph->scheduleLoadSceneFile(scenePath);
 
-    // Initialize the RenderEngine
-    _renderEngine->setSceneGraph(sceneGraph);
     _renderEngine->initialize();
+
+    scheduleLoadScene(scenePath);
+
+    LINFO("Finished initializing");
+}
+
+void OpenSpaceEngine::scheduleLoadScene(const std::string& scenePath) {
+    _switchScene = true;
+    _scenePath = scenePath;
+}
+
+void OpenSpaceEngine::loadScene(const std::string& scenePath) {
+    
+    windowWrapper().setBarrier(false);
+    windowWrapper().setSynchronization(false);
+    OnExit(
+        [this]() {
+            windowWrapper().setSynchronization(true);
+            windowWrapper().setBarrier(true);
+        }
+    );
+    
+    // Run start up scripts
+    try {
+        runPreInitializationScripts(scenePath);
+    }
+    catch (const ghoul::RuntimeError& e) {
+        LERRORC(e.component, e.message);
+    }
+
+
+    Scene* scene;
+    try {
+        scene = _sceneManager->loadScene(scenePath);
+    } catch (const ghoul::FileNotFoundError& e) {
+        LERRORC(e.component, e.message);
+        return;
+    } catch (const Scene::InvalidSceneError& e) {
+        LERRORC(e.component, e.message);
+        return;
+    } catch (const ghoul::RuntimeError& e) {
+        LERRORC(e.component, e.message);
+        return;
+    }
+
+    Scene* previousScene = _renderEngine->scene();
+    if (previousScene) {
+        _syncEngine->removeSyncables(Time::ref().getSyncables());
+        _syncEngine->removeSyncables(_renderEngine->getSyncables());
+        _syncEngine->removeSyncable(_scriptEngine.get());
+
+        _renderEngine->setScene(nullptr);
+        _renderEngine->setCamera(nullptr);
+        _sceneManager->unloadScene(*previousScene);
+    }
+
+    // Initialize the RenderEngine
+    _renderEngine->setScene(scene);
+    _renderEngine->setCamera(scene->camera());
     _renderEngine->setGlobalBlackOutFactor(0.0);
     _renderEngine->startFading(1, 3.0);
-    
+
+    scene->initialize();
+    _interactionHandler->setCamera(scene->camera());
+
+    try {
+        runPostInitializationScripts(scenePath);
+    }
+    catch (const ghoul::RuntimeError& e) {
+        LFATALC(e.component, e.message);
+    }
+
+    // Write keyboard documentation.
+    const std::string KeyboardShortcutsType =
+        ConfigurationManager::KeyKeyboardShortcuts + "." +
+        ConfigurationManager::PartType;
+
+    const std::string KeyboardShortcutsFile =
+        ConfigurationManager::KeyKeyboardShortcuts + "." +
+        ConfigurationManager::PartFile;
+
+    std::string type, file;
```

Comment:
I managed to reduce scope with putting documentation generation in 2 separate blocks.
I'd rather stick to checking for existence and retrieving in the same step. (Saving one string based lookup. Not that it matters here, but I think it's good practice anyway.)